### ISSUE TITLE
A runtime crash bug caused by analyzing a class file

### ISSUE THREAD

**Issue by kitty-1998** [soot-oss/SootUp#897]
**Bug Description**

I use SootUp to analyze a class `kotlin.reflect.jvm.internal.impl.descriptors.runtime.structure.ReflectClassUtilKt` in jar file (https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-reflect/1.9.20) and find a runtime crash. Kotlin-reflect is a popular and fundamental library for Kotlin programming language.

**The minimized example is listed below:**

The input class is attached here: 
[Class.zip](https://github.com/soot-oss/SootUp/files/14666647/Class.zip)

```Kotlin
package org.example
import kotlin.reflect.jvm.internal.impl.name.*;
val Class<*>.classId: ClassId
  get() = when {
      else -> declaringClass?.classId?.createNestedClassId(Name.identifier(simpleName)) ?: ClassId.topLevel(FqName(name))
  }
```

**SootUp Setup**
```java
AnalysisInputLocation<JavaSootClass> classInput = new JavaClassPathAnalysisInputLocation("Path_To_Compiled_Class", SourceType.Application);
JavaProject project = JavaProject.builder(new JavaLanguage(8))
          .addInputLocation(classInput)
          .addInputLocation(javaBaseInputLocation)
          .build();
JavaView view = project.createView();
```

**To reproduce**
Run SootUp with the above setup and class file.

**Stacktrace**
```bash
java.lang.RuntimeException: Failed to convert <org.example.MainKt: kotlin.reflect.jvm.internal.impl.name.ClassId getClassId(java.lang.Class)>
        at sootup.java.bytecode.frontend.AsmMethodSource.resolveBody(AsmMethodSource.java:216)
        at sootup.core.model.SootMethod.lazyBodyInitializer(SootMethod.java:98)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:186)
        at sootup.core.model.SootMethod.getBody(SootMethod.java:177)
        at sootup.callgraph.AbstractCallGraphAlgorithm.resolveAllCallsFromSourceMethod(AbstractCallGraphAlgorithm.java:197)
        at sootup.callgraph.AbstractCallGraphAlgorithm.processWorkList(AbstractCallGraphAlgorithm.java:159)
        at sootup.callgraph.AbstractCallGraphAlgorithm.constructCompleteCallGraph(AbstractCallGraphAlgorithm.java:83)
        at sootup.callgraph.ClassHierarchyAnalysisAlgorithm.initialize(ClassHierarchyAnalysisAlgorithm.java:63)
        at edu.ssrlab.analysis.SootUpExecutor.runCHA(SootUpExecutor.java:125)
        at edu.ssrlab.TestMutiFolder.testSootUpCHA(TestMutiFolder.java:70)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
......
```

**Comment by swissiety** [comment]
Hi @kitty-1998,
which version of SootUp do you use?

**Comment by kitty-1998** [comment]
> Hi @kitty-1998, which version of SootUp do you use?

Hi, @swissiety  I used the lastest version 1.1.2.

**Comment by swissiety** [comment]
Ok, please try the latest develop state - It should work with it - in other words: I was not able to reproduce the error with your input from the zip.

**Comment by JonasKlauke** [comment]
I assume that it crashes because of the empty if branch in the if operator. I dont know kotlin but "?:" is probaly the if operator or?
could you fill in something in the if branch and test it again

**Comment by kitty-1998** [comment]
Thanks for your reply, it seems that the bug has been fixed in the develop branch `5af272dcda` that was released on Apr 4.
So, I close the issue.
